From 860b6caf643e87825c0f3d0342d1b204bcd710db Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Mon, 4 Feb 2013 12:05:11 +0530
Subject: [PORT FROM MAIN] [PORT FROM R41]Both sound and time are abnormal when playback LongDuration_MIDI_5m23s file

BZ: 79222

The mDuration field in MidiFile is not updated until
getDuration is called for the first time. With this change
we are updating mDuration field during prepareAsync call.

Change-Id: I75aaf92efd41dbcfaf008cc71561ac7b0eda201b
Orig-Change-Id: I728a6dccd9687986a368eb68dfefdbe4cd57bc49
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Signed-off-by: sreekanX <sreekanthx.p.c@intel.com>
Reviewed-on: http://android.intel.com:8080/94263
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Tested-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 media/libmediaplayerservice/MidiFile.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/MidiFile.cpp b/media/libmediaplayerservice/MidiFile.cpp
index 8db5b9b..0fd8ff2 100644
--- a/media/libmediaplayerservice/MidiFile.cpp
+++ b/media/libmediaplayerservice/MidiFile.cpp
@@ -199,6 +199,13 @@ status_t MidiFile::prepareAsync()
     } else {
         sendEvent(MEDIA_ERROR, MEDIA_ERROR_UNKNOWN, ret);
     }
+    int duration;
+    // mDuration is obtained before the start() call, so that when
+    // playback is started without calling getDuration, it remains updated.
+    // mDuration is updated when getDuration() is called for the first time.
+    if (getDuration(&duration) == NO_ERROR) {
+       ALOGV("Updated mDuration inside getDuration method");
+    }
     return ret;
 }
 
-- 
1.7.4.1

